docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,196 @@
+# Semantic Conventions for ASP.NEt Core specific metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for ASP.NET Core metrics, not specific to HTTP
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Server](#server)
+- [Routing](#routing)
+  * [Metric: `aspnet.routing.match_attempts`](#metric-aspnetroutingmatch_attempts)
+- [Exception metrics](#exception-metrics)
+  * [Metric: `aspnet.diagnostics_handler.exceptions`](#metric-aspnetdiagnostics_handlerexceptions)
+- [Rate-limiting](#rate-limiting)
+  * [Metric: `aspnet.rate_limiting.active_request_leases`](#metric-aspnetrate_limitingactive_request_leases)
+  * [Metric: `aspnet.rate_limiting.request_lease.duration`](#metric-aspnetrate_limitingrequest_leaseduration)
+  * [Metric: `aspnet.rate_limiting.queued_requests`](#metric-aspnetrate_limitingqueued_requests)
+  * [Metric: `aspnet.rate_limiting.request.time_in_queue`](#metric-aspnetrate_limitingrequesttime_in_queue)
+  * [Metric: `aspnet.rate_limiting.requests`](#metric-aspnetrate_limitingrequests)
+
+<!-- tocstop -->
+
+## Server
+
+## Routing
+
+All routing metrics are reported by `Microsoft.AspNetCore.Routing` meter.
+
+### Metric: `aspnet.routing.match_attempts`
+
+<!-- semconv metric.aspnet.routing.match_attempts(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.routing.match_attempts` | Counter | `{match}` | Number of requests that were attempted to be matched to an endpoint. [1] |
+
+**[1]:** Meter name is `Microsoft.AspNetCore.Routing`.
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.routing.match_attempts(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.routing.match_status` | string | Match result - success or failure | `success`; `failure` | Recommended |
+| `aspnet.routing.fallback` | string | TODO | `TODO` | Required |
```

Comment:
```suggestion
| `aspnet.routing.route.is_fallback` | Boolean | A value that indicates whether the matched route is a fallback route. | `true`; `false` | Conditionally Required: If and only if a route was successfully matched. |
```

Reply:
we can propose attributes in http.* to otel if there are similar concepts in at least a few other frameworks. AFAIK istio folk were looking into custom routing definition, but i'd not expect it to go through before HTTP semconv stabilization.

So if .NET uses http.* now, there is a chance, it'd become part of semconv as is, as a different attribute, etc.  The worst case is if the same attribute name is used, but the semantics is different.

If you know other frameworks that support routing fallbacks and follow similar terminology, mind sharing some links? I'll create an issue in otel (or feel free to create one yourself). Otherwise, let's keep `aspnet`? E.g. `aspnet.routing.is_fallback_route` to avoid `routing.route` ?